diff --git a/compiler/libpc300/sc6.c b/compiler/libpc300/sc6.c
--- a/compiler/libpc300/sc6.c
+++ b/compiler/libpc300/sc6.c
@@ -1046,7 +1046,8 @@ static void append_dbginfo(FILE *fout)
 
   /* state table */
   for (constptr=sc_state_tab.next; constptr!=NULL; constptr=constptr->next) {
-    assert(strlen(constptr->name)>0);
+    if (strlen(constptr->name)==0)
+      continue;
     dbghdr.states++;
     dbghdr.size+=sizeof(AMX_DBG_STATE)+strlen(constptr->name);
   } /* for */
@@ -1151,7 +1152,8 @@ static void append_dbginfo(FILE *fout)
 
   /* state table */
   for (constptr=sc_state_tab.next; constptr!=NULL; constptr=constptr->next) {
-    assert(strlen(constptr->name)>0);
+    if (strlen(constptr->name)==0)
+      continue;
     id1=(int16_t)constptr->value;
     id2=(int16_t)constptr->index;
     address=(ucell)constptr->value;
